From 89a314b03d0b2e600bb503a1e3d75faa655d65e0 Mon Sep 17 00:00:00 2001
From: Moritz Haase <Moritz.Haase@bmw.de>
Date: Wed, 2 Jul 2025 08:37:21 +0200
Subject: [PATCH] cmake: Support builds with CMake 4+

CMake 4 has removed support for compatibility with CMake < 3.5 and thus builds
fail with:

    CMake Error at CMakeLists.txt:2 (cmake_minimum_required):
      Compatibility with CMake < 3.5 has been removed from CMake.

      Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
      to tell CMake that the project requires at least <min> but has been
      updated to work with policies introduced by <max> or earlier.

      Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.

To ensure builds with older CMake versions are still possible, use the suggested
'<min>...<max>' syntax instead of simply bumping '<min>'. To avoid a deprecation
warning by CMake 3.31+, 3.10 was chosen as '<max>' instead of 3.5.

While at it, move the 'cmake_minimum_required()' call before 'project()' to fix:

    CMake Warning (dev) at CMakeLists.txt:2 (project):

      cmake_minimum_required() should be called prior to this top-level
      project() call.  Please see the cmake-commands(7) manual for usage
      documentation of both commands.

Signed-off-by: Moritz Haase <Moritz.Haase@bmw.de>
Upstream-Status: Submitted [https://git.adelielinux.org/adelie/musl-locales/-/merge_requests/10]
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c8ac5c4..5744761 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
+cmake_minimum_required(VERSION 2.8...3.10)
 project(locales C)
-cmake_minimum_required(VERSION 2.8)
 
 option(LOCALE_PROFILE "Install profile file setting the MUSL_LOCPATH environment variable" ON)
 
